mono/utils/mono-mmap.c

```diff
@@ -453,6 +455,7 @@ mono_mprotect (void *addr, size_t length, int flags)
 #endif
 #endif
 	}
+	// No GC safe transition because this is called early in mini_init via mono_arch_init (with a few layers of indirection)
 	return mprotect (addr, length, prot);
```

Comment:
Would seem to be free of I/O, but involve a lock (all virtual alloc/free/protect requires synchronization, like flushing multi-processor TLBs).